include/z64player.h

```diff
@@ -188,30 +198,114 @@ typedef enum {
     /* 34 */ PLAYER_MWA_MAX
 } PlayerMeleeWeaponAnimation;
 
-typedef struct {
-    /* 0x00 */ f32 unk_00;
-    /* 0x04 */ f32 unk_04;
+typedef enum PlayerDoorType {
+    /* -1 */ PLAYER_DOORTYPE_MINUS_1 = -1, // PLAYER_DOORTYPE_AJAR = -1,
+    /*  0 */ PLAYER_DOORTYPE_NONE,
+    /*  1 */ PLAYER_DOORTYPE_HANDLE,
+    /*  2 */ PLAYER_DOORTYPE_SLIDING,
+    /*  3 */ PLAYER_DOORTYPE_3, // PLAYER_DOORTYPE_FAKE // Unused?
+    /*  4 */ PLAYER_DOORTYPE_STAIRCASE,
+    /*  5 */ PLAYER_DOORTYPE_5
```

Comment:
5 is only set in EnDoor with this condition
```c

                        player->doorType = ((this->door.dyna.actor.textId == 0x1821) && (D_80867BC0[0] != 0))
                                               ? PLAYER_DOORTYPE_5
                                               : PLAYER_DOORTYPE_MINUS_1;
```
That textid refers to
> "Ah...You are a member.
> Please come in."

And the code doesn't seem to do anything special about this case

About MINUS_1, I don't know what an "ajar" is

Reply:
ajar means slightly open. like someone opened it a bit and is peaking out

Reply:
looking closer at the oot code and the mm code, i believe it is misnamed and is really a "talking door" that doesnt let you in. oot has this misnamed cause ajar doors do talk to you as well, so whoever named it thought it was just for ajar doors

assuming that is true, and based on the context posted above, type 5 would be a talking door that still lets you in after you close the text (hence "please come in"). This should all be double checked though

Reply:
5 is specifically the Milk Bar door, then, although might change depending on ToD.

Reply:
not sure exactly whats going on with 5, but -1 is definitley talking 
```c
            if (this->doorType < 0) {
                func_8085B460(play, doorActor);
                if (doorActor->textId == 0x1821) {
                    doorActor->flags |= ACTOR_FLAG_100;
                }
                return true;
            }
  ```
  
  similar oot block 
  ```
              if (this->doorType <= PLAYER_DOORTYPE_AJAR) {
                doorActor->textId = 0xD0;
                func_80853148(play, doorActor);
                return 0;
            }
```
it still needs to be negative for it to talk to you even in the milk bar case, but maybe it changes to 5 later. needs more digging

Reply:
What about naming -1 as `PLAYER_DOORTYPE_TALK_LOCKED` and 5 as `PLAYER_DOORTYPE_TALK_TALK_OPEN`?

Reply:
type 5 would need to be looked at closer, it was only my assumption based on context that its a "talking door that lets you in"
afaict the "open" talking door will still need to be -1 to talk to you. so it may not say anything about open/close